File: app/src/main/java/com/highcom/passwordmemo/PasswordMemoLifecycle.java
package com.highcom.passwordmemo;

import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;
import androidx.lifecycle.Lifecycle;

public class PasswordMemoLifecycle implements LifecycleObserver {
    private static boolean isBackground = false;

    public static boolean getIsBackground() {
        return isBackground;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    public void onResume() {
        isBackground = false;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    public void onPause() {
        isBackground = true;
    }
}

File: app/src/main/java/com/highcom/passwordmemo/PasswordMemoLifecycle.kt
@file:Suppress("DEPRECATION")

package com.highcom.passwordmemo

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent

class PasswordMemoLifecycle : LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onResume() {
        isBackground = false
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() {
        isBackground = true
    }

    companion object {
        var isBackground = false
            private set
    }
}